pkg/sql/conn_executor_test.go

```diff
@@ -149,36 +148,43 @@ CREATE TABLE t.test (k INT PRIMARY KEY, v TEXT);
 	for _, state := range tests {
 		t.Run(state, func(t *testing.T) {
 			// Create a low-level lib/pq connection so we can close it at will.
-			pgURL, cleanupDB := sqlutils.PGUrl(
+			pgURL, cleanup := sqlutils.PGUrl(
```

Comment:
Can we call s.ApplicationLayer().PGUrl here? I think it does a very similar dance.

Reply:
Huh :)

In that case, the main motivation of this PR is already addressed. I'll keep it open until we find another case where this API is definitely needed.

Reply:
I personally still like the DeploymentMode method. I think there are cases where it would make some of the tests much cleaner.

Reply:
That sounds good. I think we are good to merge this then.